### PR TITLE
[CARBONDATA-3380] Fix missing appName and AnalysisException bug in DirectSQLExample

### PR DESCRIPTION
**Pull request by lamber-ken**
Fix two bug in DirectSQLExample
- fix missing appName which is writing the carbondata files.
- fix AnalysisException bug because of carbonfile
<br>

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed? NO
 
 - [x] Any backward compatibility impacted? NO
 
 - [x] Document update required? NO

 - [x] Testing done YES
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required? It's a example.
        - How it is tested? Please attach test report. 
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3164/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11429/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3362/

**Comment by lamber-ken** [comment]
Hi, @xubo245, can you please take a look?

**Comment by xubo245** [comment]
Please update the PR content. @lamber-ken

**Comment by lamber-ken** [comment]
> Please update the PR content. @lamber-ken

Thanks for your comment, but I can't get your point.

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/DirectSQLExample.scala [hunk 1]
```diff
@@ -86,7 +87,7 @@ object DirectSQLExample {
     println("Running SQL on carbon files directly")
     try {
       // 2. run queries directly, no need to create table first
-      sql(s"""select * FROM  carbonfile.`$readPath` limit 10""".stripMargin).show()
+      sql(s"""select * FROM carbon.`$readPath` limit 10""".stripMargin).show()
```
Comment (xubo245) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/DirectSQLExample.scala]: Can you add this example(DirectSQLExample) to org.apache.carbondata.examplesCI.RunExamples? and do some check? which can avoid fail when other developer change the code in the future.

Comment (lamber-ken) [reply examples/spark2/src/main/scala/org/apache/carbondata/examples/DirectSQLExample.scala]: Done

**Comment by lamber-ken** [comment]
retest this please.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3359/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3357/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3559/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11626/

**Comment by lamber-ken** [comment]
hi, @xubo245 , the qa gone through, do you have time to check again? thanks.

**Comment by xubo245** [comment]
LGTM

**Comment by xubo245** [comment]
@zzcclp Can you help to review it?

**Comment by lamber-ken** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3609/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11883/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3817/

**Comment by xubo245** [comment]
@lamber-ken  Why it failed?

**Comment by xubo245** [comment]
@zzcclp @kunal642 Can you help to review it?

**Comment by lamber-ken** [comment]
> @lamber-ken Why it failed?

hi @xubo245 , I'll retest it again

**Comment by lamber-ken** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3626/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11900/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3833/

File: examples/spark2/src/test/scala/org/apache/carbondata/examplesCI/RunExamples.scala [hunk 2]
```diff
@@ -129,4 +129,8 @@ class RunExamples extends QueryTest with BeforeAndAfterAll {
   test("HiveExample") {
     HiveExample.exampleBody(spark, TestQueryExecutor.warehouse)
   }
+
+  test("DirectSQLExample") {
+    DirectSQLExample.main(null)
```
Comment (zzcclp) [inline examples/spark2/src/test/scala/org/apache/carbondata/examplesCI/RunExamples.scala]: Please refer to 'CarbonSessionExample.exampleBody(spark)', use 'spark' created in RunExamples, don't re-create one in DirectSQLExample.

Comment (lamber-ken) [reply examples/spark2/src/test/scala/org/apache/carbondata/examplesCI/RunExamples.scala]: Thanks for your suggesion @zzcclp, done.

**Comment by lamber-ken** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3639/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3641/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11915/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3845/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3850/

**Comment by lamber-ken** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3644/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11918/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3854/

**Comment by lamber-ken** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3647/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3857/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11921/

**Comment by lamber-ken** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3650/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11924/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3860/

**Comment by zzcclp** [comment]
@xubo245 please review again, thanks.

**Comment by xubo245** [comment]
LGTM

**Comment by zzcclp** [comment]
OK, I will merge this pr after retest successfully.

**Comment by zzcclp** [comment]
retest this please.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3687/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11962/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3897/

**Pull request closed without merging**